search/includes/classes/class-search.php

```diff
@@ -531,8 +531,6 @@ public function filter__ep_bulk_items_per_page() {
 	}
 
 	public function filter__ep_do_intercept_request( $request, $query, $args, $failures ) {
-		$fallback_error = new \WP_Error( 'vip-search-upstream-request-failed', 'There was an error connecting to the upstream search server' );
```

Comment:
@nickdaugherty I'm not sure why this is here, and whether removing it could lead to any unexpected problem

Reply:
@netsuso this is the value that is returned to the calling code from `vip_safe_wp_remote_request()` if the request is not successful. We need to keep it so that we correctly bubble up a WP_Error() to calling code, to match the behavior of the WP_HTTP API, as lots of code checks for errors with `is_wp_error()`.

Reply:
Discussed more on Slack - the intent of the question was on if we can use the default WP_Errors that get returned, so we have more granularity in the logging, which would be good. Then we can turn those into a more user-friendly error that gets returned for surfacing higher up.

Reply:
I've updated the PR according to that discussion